src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -806,8 +869,66 @@ def {_DEFAULT_HANDLER_NAME}({wrapper_params}):
 """.strip()
 
 
+def add_snowpark_package_to_sproc_packages(
+    session: Optional["snowflake.snowpark.Session"],
+    packages: Optional[List[Union[str, ModuleType]]],
+) -> List[Union[str, ModuleType]]:
+    major, minor, patch = VERSION
+    package_name = "snowflake-snowpark-python"
+    # Use == to ensure that the remote version matches the local version
+    this_package = f"{package_name}=={major}.{minor}.{patch}"
+
+    # When resolve_imports_and_packages is called below it will use the provided packages or
+    # default to the packages in the current session. If snowflake-snowpark-python is not
+    # included by either of those two mechanisms then create package list does include it and
+    # any other relevant packages.
+    if packages is None:
+        if session is None:
+            packages = [this_package]
+        elif package_name not in session._packages:
+            packages = list(session._packages.values()) + [this_package]
+    else:
+        if not any(package_name in p for p in packages):
+            packages.append(this_package)
+    return packages
+
+
+def resolve_packages_in_sandbox(
+    packages: Optional[List[Union[str, ModuleType]]],
+) -> List[str]:
+    """
+    Special function invoked only when executing Snowpark code in a sandbox environment.
```

Comment:
The word sandbox is an existing term in the backend, let's change this to something like native_apps_cli?

Reply:
Discussed on zoom, `native_apps_cli` is too specific, this is any generic sandbox that is created for a client running the library to prevent talking to snowflake. Changed it accordingly in context.py and elsewhere.